src/lib/parsers/sol-json.c

```diff
@@ -282,6 +282,9 @@ sol_json_token_get_double(const struct sol_json_token *token, double *value)
     char *endptr;
     int r;
 
+    SOL_NULL_CHECK(token, -EINVAL);
```

Comment:
Since these parameters are more likely to be allocated on the stack (and thus their addresses are guaranteed non-null), adding this overhead (as cheap as it is) is not the best course of action here. Use `SOL_ATTR_NONNULL` instead: this will generate a compile-time warning.